### ISSUE TITLE
$XDG_DATA_HOME/Steam is not used as Steam installation on Debian-based systems

### ISSUE THREAD

**Issue by kakurasan** [lhark/truckersmp-cli#32]
`$XDG_DATA_HOME/Steam/config/loginusers.vdf` is not created/used on Debian-based systems if their `steam` package is used.

The Steam installation directory is:

* `~/.steam/debian-installation` on recent Debian (e.g. Buster)
* `~/.steam` on older Debian or recent Ubuntu (e.g. Ubuntu 19.04)

The `steam` command is a shell script (`/usr/games/steam`) but it's difficult to detect the actual installation directory.

The old versions of the package (e.g. `1.0.0.54-2`) have `config=$HOME/.steam` line and `~/.steam` is always used, but the newer versions (e.g. `1.0.0.59-4`) use `~/.steam` or `~/.steam/debian-installation`.

* [Older: 1.0.0.54-2](http://deb.debian.org/debian/pool/non-free/s/steam/steam_1.0.0.54-2.debian.tar.xz)
* [Newer: 1.0.0.59-4](http://deb.debian.org/debian/pool/non-free/s/steam/steam_1.0.0.59-4.debian.tar.xz)

On my Ubuntu 19.04 system,

* `~/.steam` is actual directory and used by Steam (Ubuntu package).
* `~/.steam/debian-installation` doesn't exist.
* `~/.local/share/Steam` doesn't exist.

**Comment by Lucki** [comment]
Original for comparison: [1.0.0.59.tar.gz](repo.steampowered.com/steam/pool/steam/s/steam/steam_1.0.0.59.tar.gz)

While the original is quite clear it seems Debian is going it's own way here although they're commenting the intended behavior at the top:

debian:
~~~ bash
# According to Valve, ~/.steam is intended to be a control directory containing
# symbolic links pointing to the currently-running or most-recently-run Steam
# installation. This is part of Steam's API, and is relied on by external
# components.
[…]
STEAMCONFIG="$HOME/.steam"
[…]
# STEAMDIR points to the actual installation root: the equivalent of
# C:\Program Files\Steam in the Windows Steam client. To avoid filename
# collisions this should be distinct from ~/.steam.
#
# The shell variable name STEAMDIR matches what's used in Valve's
# /usr/bin/steam.
[…]
    # The historical Debian behaviour has been to use ~/.steam as the
    # installation directory in addition to using it as the control
    # directory.This causes some file collisions, so we've moved away
    # from that, but we can't easily disentangle this in existing
    # installations.
    STEAMDIR="$HOME/.steam"
else
    # This is a new installation, so use a distinct directory to avoid
    # file collisions. Valve would use $XDG_DATA_HOME/Steam here.
    # Debian uses a subdirectory of ~/.steam, to avoid having a mixture
    # of XDG basedirs and traditional dotfiles in the same application.
    STEAMDIR="$HOME/.steam/debian-installation"
~~~

original:
~~~ bash
function setup_variables()
{
	[…]
	STEAMCONFIG=~/.steam
	[…]
	# Get the default data path
	STEAM_DATA_HOME="${XDG_DATA_HOME:-$HOME/.local/share}"
	case "$STEAMPACKAGE" in
		steam)
			CLASSICSTEAMDIR="$HOME/Steam"
			DEFAULTSTEAMDIR="$STEAM_DATA_HOME/Steam"
~~~

**Comment by kakurasan** [comment]
Fixed in 8aa88ca, these Steam installation directories can be detected/used now.
